### DISCUSSION TITLE
UWP, WinUI3 and Inbox apps

### DISCUSSION THREAD

**Discussion by memeplex** [microsoft/WindowsAppSDK#1070] [Q&A]
Hi all, I'm a bit confused and curious about some issues:

1. What technology are the apps provided with Windows 10/11 currently using? I assume some of the inbox apps use UWP, but older or bigger apps are probably Win32 ones.

2. Since they are being modernized in Windows 11, how are the Win32 ones being refreshed? Take for example the File Explorer, or even Office. WinUI 2 islands?

3. For the UWP ones, what will happen when WinUI 3 becomes mainstream since UWP support is not planned for it? Are they being rewritten as PWA or Win32 maybe?

Thank you in advance

Carlos

**Comment by driver1998** [comment]
1. Most of the "new" (added on Windows 10 or later) ones are UWP, but some uses different UI stack. Power Automate uses WPF for example, and Windows Terminal is Win32 + XAML Islands.
2. Like Windows Terminal, they are Win32 apps with (sometimes multiple) XAML Islands attached. Most of the time it is System XAML Island with WinUI 2. \
Except File Explorer, which is a mix of System XAML and WinUI 3 XAML (Don't ask me how it is done, it is a giant hack, and not documented at all). The taskbar uses System XAML, but folder windows uses WinUI 3. 
3. This is the plan, Microsoft provides guidance on migrating from UWP to WinUI 3 on Desktop (aka Win32 app model).

**Reply by ghost1372** [reply]
Action center moved from uwp to win32 + xaml and not winui 3
